src/network/networkd-ndisc.c

```diff
@@ -612,6 +612,11 @@ static int ndisc_router_process_route(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Failed to get route prefix length: %m");
 
+        if (!link->network->ipv6_accept_ra_use_gateway &&
```

Comment:
Hmm, in `ndisc_router_process_default()`:

https://github.com/systemd/systemd/blob/a420d71793bcbc1539a63be60f83cdc14373ea4a/src/network/networkd-ndisc.c#L346-L363

So do we even need `!link->network->ipv6_accept_ra_use_gateway &&` here?

Reply:
When networkd receives an RA, then networkd creates the default gateway via the router address. Also, networkd creates routes defined in route prefixes. If an RA contains a route prefix whose destination is ::/0, then the created route becomes the default gateway. Hence, if `UseGateway=no` is set, we also need to check route prefixes contained in RA.

Reply:
> If an RA contains a route prefix whose destination is ::/0, then the created route becomes the default gateway.

And the nexthop/via of that will always also be the router address? If that's the case, `ndisc_router_process_route()` can return (0) whenever `in6_addr_is_null(&dst) && prefixlen == 0`, regardless of whether `link->network->ipv6_accept_ra_use_gateway` is false?

Reply:
Indeed. Updated.